### PR TITLE
Release 0.3.0

### PR DESCRIPTION
**Pull request by JinmingHu-MSFT**
1. Return error information when libcurl fails, fix https://github.com/Azure/azure-storage-cpplite/issues/62

2. Add sample for `blob_client`, as proposed in https://github.com/Azure/azure-storage-cpplite/issues/68

3. Change default retry count, so that people won't wrongly think the program hangs when it's actually retrying. https://github.com/Azure/azure-storage-cpplite/issues/65, https://github.com/Azure/azure-storage-cpplite/issues/61

4. Proxy support, requested in https://github.com/Azure/azure-storage-cpplite/issues/71

5. Fix a bug: error when downloading empty blob

6. Fix authentication failure when running against Azurite and IoT devices, https://github.com/Azure/azure-storage-cpplite/issues/66, https://github.com/Azure/azure-storage-cpplite/issues/24

7. Fix a bug: duplicate http headers on retry

8. Parallel uploading & downloading

9. Use toolset v140 on VS2017 to replace VS2015, since Microsoft-hosted VS2015 image on Azure Pipelines is being removed.

10. Add Win32 configuration to Azure Pipelines

11. Change the size parameter in upload/download block/blob APIs from `size_t` to `uint64_t`

### CODE REVIEW

**Comment by JinmingHu-MSFT** [comment]
The auzre pipelines test on VS2015 failed because latest vcpkg doesn't compile. Not our problem.

File: include/executor.h [hunk 1]
```diff
@@ -76,8 +76,18 @@ namespace azure {  namespace storage_lite {
                         std::string str(std::istreambuf_iterator<char>(s.istream()), std::istreambuf_iterator<char>());
```
Comment (katmsft) [inline include/executor.h]: Is there a reason that this is not consistent with the code below? I think str is also not used when code is not CURLE_OK.
Also, I think the two if statement can be merged.
Both are minor comments, consider this already signed-off with some minor suggestions :)

Comment (JinmingHu-MSFT) [reply include/executor.h]: > Is there a reason that this is not consistent with the code below? I think str is also not used when code is not CURLE_OK.

It's used in `else if` clause.

Comment (JinmingHu-MSFT) [reply include/executor.h]: And what do you mean `two if statement can be merged`, I don't see that'll simplify code logic.

File: include/mstream.h [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                else if (dir == std::ios_base::end)
+                {
+                    m_cur = m_end + off;
+                }
+                else if (dir == std::ios_base::beg)
+                {
+                    m_cur = m_data + off;
+                }
+            }
+            return m_cur - m_data;
```
Comment (katmsft) [inline include/mstream.h]: Is it by design to enable seek to an out-of-range position?

Comment (JinmingHu-MSFT) [reply include/mstream.h]: No, out-of-range seeking is undefined bahavior.

**Pull request merged**